cdc/entry/mounter.go

```diff
@@ -314,16 +383,94 @@ func datum2Column(tableInfo *model.TableInfo, datums map[int64]types.Datum, fill
 			// ApproximateBytes = column data size + column struct size
 			ApproximateBytes: colSize + sizeOfEmptyColumn,
 		}
+		columnIDs = append(columnIDs, colInfo.ID)
+	}
+	return cols, rawCols, columnIDs, nil
+}
+
+// return error if cannot get the expected checksum from the decoder
+// return false if the checksum is not matched
+// return true if the checksum is matched and the checksum is the matched one.
+func (m *mounter) verifyChecksum(
+	columnIDs []int64, rawColumns []types.Datum, isPreRow bool,
+) (uint32, bool, error) {
+	if !m.integrity.Enabled() {
+		return 0, true, nil
+	}
+
+	// when the old value is not enabled, no previous columns so that no need to verify the checksum.
+	if isPreRow && !m.enableOldValue {
+		return 0, true, nil
 	}
-	return cols, rawCols, nil
+
+	var decoder *rowcodec.DatumMapDecoder
+	if isPreRow {
+		decoder = m.preDecoder
+	} else {
+		decoder = m.decoder
+	}
+	if decoder == nil {
```

Comment:
seems like this will never happen? can use panic instead of error here, which will make debugging easier.

Reply:
I would prefer error, to avoid one single changefeed makes the whole server crashed